### ISSUE TITLE
Handle duplicated packets

### ISSUE THREAD

**Issue by PurpleMyst** [connorkuehl/tftp#46]
Per the title, we should handle duplicate packets such as receiving two Data packets with the same block id or two Ack packets with the same block id. (by silently dropping it, as far as I can see?)

Furthermore, getting two responses for a read/write request should somehow be handled, per the end of §4:

> Host A sends a request to host B. Somewhere in the network, the request packet is
   duplicated, and as a result two acknowledgments are returned to host
   A, with different TID's chosen on host B in response to the two
   requests.  When the first response arrives, host A continues the
   connection.  When the second response to the request arrives, it
   should be rejected, but there is no reason to terminate the first
   connection.  Therefore, if different TID's are chosen for the two
   connections on host B and host A checks the source TID's of the
   messages it receives, the first connection can be maintained while
   the second is rejected by returning an error packet.

**Comment by PurpleMyst** [comment]
This will be trivial to bolt onto #6's PR I'm writing

**Comment by connorkuehl** [comment]
Good catch, @PurpleMyst! 

> The former will be trivial to bolt onto #6's PR I'm writing

If you don't mind, I'd prefer each feature/fix gets its own separate PR.

> Furthermore, getting two responses for a read/write request should somehow be handled, per the end of §4:
> 
> > Host A sends a request to host B. Somewhere in the network, the request packet is
> > duplicated, and as a result two acknowledgments are returned to host
> > A, with different TID's chosen on host B in response to the two
> > requests.  When the first response arrives, host A continues the
> > connection.  When the second response to the request arrives, it
> > should be rejected, but there is no reason to terminate the first
> > connection.  Therefore, if different TID's are chosen for the two
> > connections on host B and host A checks the source TID's of the
> > messages it receives, the first connection can be maintained while
> > the second is rejected by returning an error packet.

I think the implementation for `client::Client` already does this by accident, but `server::Server` will need to support this behavior. I'll file an issue.

**Comment by PurpleMyst** [comment]
> If you don't mind, I'd prefer each feature/fix gets its own separate PR.
That's what I meant :)

**Comment by PurpleMyst** [comment]
The former still stands though, we shouldn't return an error packet for two ACKs. Once the PR for #6 is merged I'll make another one that addresses that, as the fix is easy to add to that code but conflicts with it if done separately
